packages/core/src/amazonqFeatureDev/util/files.ts

```diff
@@ -39,7 +39,16 @@ export async function prepareRepoData(
         const ignoredExtensionMap = new Map<string, number>()
 
         for (const file of files) {
-            const fileSize = (await fs.stat(file.fileUri)).size
+            let fileSize
+            try {
+                fileSize = (await fs.stat(file.fileUri)).size
+            } catch (error) {
+                if (error instanceof Error && error.message.includes('ENOENT')) {
```

Comment:
use `hasCode` and check the `code`.  https://github.com/aws/aws-toolkit-vscode/blob/3d9077264f9d40d7e80383c3c1e5a4fd5e2181a9/packages/core/src/shared/errors.ts#L602